src/github.com/getlantern/flashlight/proxied/proxied.go

```diff
@@ -154,11 +163,16 @@ func (df *dualFetcher) RoundTrip(req *http.Request) (*http.Response, error) {
 // header to specify the fronted URL to use.
 func (df *dualFetcher) do(req *http.Request, chainedFunc func(*http.Request) (*http.Response, error), ddfFunc func(*http.Request) (*http.Response, error)) (*http.Response, error) {
 	log.Debugf("Using dual fronter")
-	frontedURL := req.Header.Get("Lantern-Fronted-URL")
-	req.Header.Del("Lantern-Fronted-URL")
+
+	op := ops.Enter("dualfetcher").Request(req)
+	defer op.Exit()
+
+	parallel := df.cf.parallel
+	frontedURL := req.Header.Get(lanternFrontedURL)
+	req.Header.Del(lanternFrontedURL)
 
 	if frontedURL == "" {
-		return nil, errors.New("Callers MUST specify the fronted URL in the Lantern-Fronted-URL header")
+		return nil, op.Error(errors.New("Callers MUST specify the fronted URL in the Lantern-Fronted-URL header"))
```

Comment:
Ok, I see. `op.Error()` is not to construct an error object, but to mark the op as failed. I suggest to rename it to `op.Failed()` and without chaining. Reducing the mental burden is way more important than saving lines of code.